.clang-tidy

```diff
@@ -0,0 +1,44 @@
+---
+Checks:
+  -*
+  ,boost-use-to-string
+  ,misc-string-compare
+  ,misc-uniqueptr-reset-release
+  ,modernize-deprecated-headers
+  ,modernize-make-shared
+  ,modernize-use-bool-literals
+  ,modernize-use-equals-delete
+  ,modernize-use-nullptr
+  ,modernize-use-override
+  ,performance-unnecessary-copy-initialization
+  ,readability-container-size-empty
+  ,readability-redundant-string-cstr
+  ,readability-static-definition-in-anonymous-namespace
+  ,readability-uniqueptr-delete-release
+WarningsAsErrors: ''
+HeaderFilterRegex: ''
```

Comment:
This should be set to `$CMSSW_BASE/src/.*` - but I don't think `clang-tidy` supports environment variables in `/clang-tidy` .

We could let `scram` adjust location, but people would likely commit that change along with theirs, which we do not want.

Maybe we should actually move `.clang-tidy` under `$CMSSW_BASE` and outside of git ?
`scram project` could create it (based on some template) with the correct  location for `HeaderFilterRegex` .
Maybe we should move the file outside of the CMSSW git repository, and let `scram project` add it to `$CMSSW_BASE` rather than `$CMSSW_BASE/src` ?